### ISSUE TITLE
Annotation Processor (For Obfuscation Step)

### ISSUE THREAD

**Issue by Zidane** [SpongePowered/Mixin#4]
@Mumfrey 

My mod Almura is basically ready for primetime and while mixins are not 100% needed, it does let some of what we are doing work 100% correctly.

Are we any further with getting this working?

**Comment by Mumfrey** [comment]
Since injection is working I'm working on it this week while I'm off work. For the time being you can just obfuscate your mixins by hand if you need them in your other project.

**Comment by Mumfrey** [comment]
So the AP is now mostly complete and generates the '*hard*' SRG mappings for SS and the '*soft*' remappings for string-based runtime member lookups as discussed. As we discussed yesterday I've now made it a hard requirement that ```@At``` "target" specifiers must be fully-qualified references, at least for now.

* Note that this hasn't undergone *any* testing to speak of so far so I fully expect stuff to break, but I'm now going out and won't have any more time to look at it until tomorrow.

The AP generates two files. One of which (```mixins.refmap.json```) should be packaged in the final jar, the other one (```mixins.srg```, name configurable via the AP arg ```reobfSrgFile``` (see [build.gradle](https://github.com/SpongePowered/Sponge/blob/master/build.gradle#L153))) contains the SRG mappings for SS.

To pass the generated SRGs to SS I need @AbrarSyed to take a look at this and figure out the best way to get the generated file into SS, like I said I won't be around tonight but I will be tomorrow, so if he sees this and has a chance to at least have a think about it that would be cool. If anyone sees him on IRC then link him to this post.

Hopefully this doesn't break things on ```Sponge```'s end, I can't test it at the moment because the source tree doesn't build :/
